### ISSUE TITLE
Button release on gamepad intermittently not captured when analog stick is in use

### ISSUE THREAD

**Issue by Godont** [godotengine/godot#45935]
**Godot version:**
v3.2.1.stable.official

**OS/device including version:**
xbox 360 controller or switch pro controller on Windows 10

**Issue description:**
When releasing a gamepad button, the `event.is_action_released` event is not always captured in `_unhandled_event` if the analog stick is also in use.

**Steps to reproduce:**

With a connected gamepad controller, "charge" the weapon by pressing the "shoot" button (A on a switch pro controller) and "fire" by releasing. There are visual indicators that this is working as expected.

However, if the analog stick is engaged for aiming (there are visual indicators), releasing the "shoot" button will sometimes fail to "fire" because `event.is_action_released("shoot")` is not triggering.

**Minimal reproduction project:**
[mwe.zip](https://github.com/godotengine/godot/files/5974220/mwe.zip)

**Comment by akien-mga** [comment]
Can you try with a more recent Godot version? 3.2.3.stable or ideally [3.2.4 RC 2](https://godotengine.org/article/release-candidate-godot-3-2-4-rc-2).

**Comment by Godont** [comment]
So I did try with 3.2.4 RC 2 and the issue is not there, however it seemed to introduce another issue at first (the label never showed "FIRED"). I'm guessing the root cause is that the `_unhandled_input` function is running at least twice between calls to `_physics_process`, because moving the code to reset the boolean tracking whether the player has fired to the firing function itself resolved that issue. In any case, the original issue seems to be resolved in this version.

Was this a known issue that got fixed, or did one of the recent changes resolve this incidentally?
